ledger-api/grpc-definitions/com/daml/ledger/api/v2/update_service.proto

```diff
@@ -0,0 +1,146 @@
+  // For transactions it lists only creates and archives, but not other events.
+  // Omits all events on transient contracts, i.e., contracts that were both created and archived in the same transaction.
+  rpc GetUpdates (GetUpdatesRequest) returns (stream GetUpdatesResponse);
+
+  // Read the ledger's complete transaction tree stream and related reassignments for a set of parties.
+  // The stream will be filtered only by the parties as wildcard parties.
+  // The template/interface filters describe the respective fields in the ``CreatedEvent`` results.
+  rpc GetUpdateTrees (GetUpdatesRequest) returns (stream GetUpdateTreesResponse);
+
+  // Lookup a transaction tree by the ID of an event that appears within it.
+  // For looking up a transaction instead of a transaction tree, please see GetFlatTransactionByEventId
+  rpc GetTransactionTreeByEventId (GetTransactionByEventIdRequest) returns (GetTransactionTreeResponse);
+
+  // Lookup a transaction tree by its ID.
+  // For looking up a transaction instead of a transaction tree, please see GetFlatTransactionById
+  rpc GetTransactionTreeById (GetTransactionByIdRequest) returns (GetTransactionTreeResponse);
+
+  // Lookup a transaction by the ID of an event that appears within it.
+  rpc GetTransactionByEventId (GetTransactionByEventIdRequest) returns (GetTransactionResponse);
+
+  // Lookup a transaction by its ID.
+  rpc GetTransactionById (GetTransactionByIdRequest) returns (GetTransactionResponse);
+}
+
+message GetUpdatesRequest {
+
+  // Beginning of the requested ledger section.
+  // The response will only contain transactions whose offset is strictly greater than this.
+  // Required
+  com.daml.ledger.api.v1.LedgerOffset begin_exclusive = 1;
+
+  // End of the requested ledger section.
+  // The response will only contain transactions whose offset is less than or equal to this.
+  // Optional, if not set, the stream will not terminate.
+  com.daml.ledger.api.v1.LedgerOffset end_inclusive = 2;
+
+  // Requesting parties with template filters.
+  // Template filters must be empty for GetUpdateTrees requests.
+  // Required
+  TransactionFilter filter = 3;
+
+  // If enabled, values served over the API will contain more information than strictly necessary to interpret the data.
+  // In particular, setting the verbose flag to true triggers the ledger to include labels, record and variant type ids
+  // for record fields.
+  // Optional
+  bool verbose = 4;
+}
+
+message GetUpdatesResponse {
+
+  // The update that matches the filter in the request.
+  Update updates = 1;
+}
+
+message Update {
+
+  oneof update {
+    Transaction transaction = 1;
+    Reassignment reassignment = 2;
+  }
+}
+
+message GetUpdateTreesResponse {
+
+  // The update that matches the filter in the request.
+  UpdateTree updates = 1;
+}
+
+message UpdateTree {
+
+  oneof update {
+    TransactionTree transaction_tree = 1;
+    Reassignment reassignment = 2;
+  }
+}
```

Comment:
Good point, thanks @stefanobaghino-da ! with 1) and 2) I agree completely (did it also)
On 3):
* I am not entirely sure about the "need to" part, since we are aiming something like a "Virtual shared ledger". On the other hand AFAIR we decided while back that we did not wanted to provide server side filtration at the moment (arguments AFAIR:the virual shared ledger argument; CN needs the merged streams; not dive deep into a discussion/design where a tranfer belongs to, which streams/ACS such filtration would make sense etc; lack of understanding whether this is something is needed at all; inefficient trivial filtration; trivial filtration is available for the users anyway easily)
* On adding the domain_id to the top level wrapper: I kind of like the idea to introduce the domain of sequencing, but we have this in each message type, where it is needed too, so it would be a redundancy. On this I have no strong opinion, we could introduce it!

@rgugliel-da @cocreature WDYT?

Reply:
> On the other hand AFAIR we decided while back that we did not wanted to provide server side filtration at the moment

To clarify, I'm not talking about server-side filtering, but client-side filtering.

> On adding the domain_id to the top level wrapper: I kind of like the idea to introduce the domain of sequencing, but we have this in each message type, where it is needed too, so it would be a redundancy. On this I have no strong opinion, we could introduce it!

I missed the fact that this was already include in individual messages, thanks for pointing that out. If that's included already, then I agree there's no reason to include it. I think for the time being I'll include it in the top-level message to avoid having to depend on changes in the digital-asset/daml repo, but changing it in the future to use the domain ID in the individual messages shouldn't be too difficult.

Thanks!

Reply:
As discussed we could add the sequencing domain_id later in a backwards compatible way.